ops/ops/interface_aws/requires.py

```diff
@@ -168,9 +176,14 @@ def region(self):
 
     @property
     def _expected_hash(self):
-        return sha256(
-            json.dumps(dict(self._to_publish), sort_keys=True).encode("utf8")
-        ).hexdigest()
+        def from_json(s: str):
+            try:
+                return json.loads(s)
+            except json.decoder.JSONDecodeError:
+                return s
+
```

Comment:
@mateoflorido @eaudetcobello
the reactive relation handler automatically encoded/decoded relation data as json if it was a complex object (dict, list ...etc).  Since ops doesn't do this -- I have to manually handle data that may or may not be json data when calculating the hash to make sure the far end integrator and remote end ops client come up with the same hash for the data.